src/Moq/ExpressionExtensions.cs

```diff
@@ -294,6 +294,16 @@ void Split(Expression e, out Expression r /* remainder */, out InvocationShape p
 					{
 						var memberAccessExpression = (MemberExpression)e;
 						Debug.Assert(memberAccessExpression.Member is PropertyInfo);
+
+						if (IsResult(memberAccessExpression.Member, out var awaitableFactory))
+						{
+							Split(memberAccessExpression.Expression, out r, out p);
+							p.AddResultExpression(
+								awaitable => Expression.MakeMemberAccess(awaitable, memberAccessExpression.Member),
+								awaitableFactory);
```

Comment:
Once we add support for custom awaitable types, it will be better to preserve the original expression, instead of replacing the `.Result` access with a new result expression.